castanet/counter.py

```diff
@@ -0,0 +1,149 @@
+"""This module counts instances of statements in Python files."""
+from typing import Dict
+import libcst.matchers as match
+
+def match_imports(cast_dict):
+    """A function for counting the number of if statements in a Python program."""
+    imports_dictionary = {}
+    # Iterate through all of the Python files in a directory
+    for file in cast_dict:
+        # Find CASTs for each of these files
+        cast = cast_dict[file]
+        # Determine number of import statements for each file
+        imports_list = match.findall(cast, match.While())
+        imports_dictionary[file] = len(imports_list)
+
+    return imports_dictionary
+
+
+def total_imports(imports_dict):
+    """Find and combine the number of import statements in Python files in a specific directory."""
+    total = 0
+    for file in imports_dict:
+        amount_of_imports = imports_dict[file]
+        total += amount_of_imports
+
+    return total
+
+
+def match_comment(cast_dict):
+    """A function for counting the number of comments statements in a Python program."""
+    comment_dictionary = {}
+    for file in cast_dict:
+        # Find CASTs for each of these files
+        cast = cast_dict[file]
+        # Determine number of comments for each file
+        comment = match.findall(cast, match.Comment())
+        comment_dictionary[file] = len(comment)
+
+    return comment_dictionary
+
+
+def total_comment(cast_dict):
+    """Add the total number of comments in directory."""
+    count = 0
+    for file in cast_dict:
+        comments = cast_dict[file]
+        count += comments
+    return count
+
+
+def count_whileloops(cast_dict):
+    """A function for counting the number of while loops in a Python program."""
+    while_dictionary = {}
+    # Iterate through all of the Python files in a directory
+    for file in cast_dict:
+        # Find CASTs for each of these files
+        cast = cast_dict[file]
+        while_loops = match.findall(cast, match.While())
+        while_dictionary[file] = len(while_loops)
+
+    return while_dictionary
+
+
+def count_forloops(cast_dict):
+    """A function for counting the number of while loops in a Python program."""
+    for_dictionary = {}
+    # Iterate through all of the Python files in a directory
+    for file in cast_dict:
+        # Find CASTs for each of these files
+        cast = cast_dict[file]
+        # Determine number of for statements for each file
+        imports = match.findall(cast, match.For())
+        for_dictionary[file] = len(imports)
+
+    return for_dictionary
+
+
+def amount_loops(loops_dict):
```

Comment:
Might `total_loops` be a better name to be more consistent with the rest of the functions?